co-log-core/src/Colog/Core/Severity.hs

```diff
@@ -0,0 +1,24 @@
+module Colog.Core.Severity
+       ( Severity (..)
+       , filterSeverity
+       ) where
+
+import Colog.Core.Action (LogAction (..), cfilter)
+
+-- | Severity for the log messages.
+data Severity
+    -- | Information useful for debug purposes
+    = Debug
+    -- | Normal operational information
+    | Info
+    -- | Important (more than average) but not critical information
+    | Notice
+    -- | General warnings
+    | Warning
+    -- | General errors/severe errors
+    | Error
+    deriving (Show, Read, Eq, Ord, Enum, Bounded)
```

Comment:
I really like that we have all these derivings! This will make `Severity` data type easier to use with `relude` and functions like `universe` and `inverseMap` 👍 

Let's also add `Ix` to `deriving`. Just in case:

* https://hackage.haskell.org/package/base-4.11.1.0/docs/Data-Ix.html